go/vt/vttest/local_cluster.go

```diff
@@ -74,10 +97,14 @@ func NoStderr() VitessOption {
 // SchemaDirectory is used to specify a directory to read schema from.
 // It cannot be used at the same time as Schema.
 func SchemaDirectory(dir string) VitessOption {
+	if dir == "" {
+		log.Fatal("BUG: provided directory must not be empty")
+	}
+
 	return VitessOption{
 		beforeRun: func(hdl *Handle) error {
-			if dir != "" {
-				hdl.cmd.Args = append(hdl.cmd.Args, "--schema_dir", dir)
```

Comment:
Good catch!

`hdl.schemaDir = dir` was missing here and this code was not covered before.

I've changed the `TestMySQL` test to use this option. I would fail now without the fix.